utils/src/set.rs

```diff
@@ -457,6 +485,229 @@ impl<K, V> DoubleEndedIterator for OrderedAssociatedIntoIter<K, V> {
     }
 }
 
+/// An ordered, deduplicated slice of items each paired with some associated value, where values must be unique.
```

Comment:
`keys` instead of `items`?
`unique` instead of `deduplicated`?
`indexed` instead of `ordered`? (I think ordered implies that they are `Ord` and are sorted)